neo/io/neomatlabio.py

```diff
@@ -199,131 +243,144 @@ def __init__(self, filename=None):
         """
         Arguments:
-            bl: the block to b saved
+            bl: the block to be saved
         """
         import scipy.io
+
         bl_struct = self.create_struct_from_obj(bl)
 
         for seg in bl.segments:
             seg_struct = self.create_struct_from_obj(seg)
-            bl_struct['segments'].append(seg_struct)
-
-            for anasig in seg.analogsignals:
-                anasig_struct = self.create_struct_from_obj(anasig)
-                seg_struct['analogsignals'].append(anasig_struct)
-
-            for irrsig in seg.irregularlysampledsignals:
-                irrsig_struct = self.create_struct_from_obj(irrsig)
-                seg_struct['irregularlysampledsignals'].append(irrsig_struct)
-
-            for ea in seg.events:
-                ea_struct = self.create_struct_from_obj(ea)
-                seg_struct['events'].append(ea_struct)
-
-            for ea in seg.epochs:
-                ea_struct = self.create_struct_from_obj(ea)
-                seg_struct['epochs'].append(ea_struct)
+            bl_struct["segments"].append(seg_struct)
+
+            for container_name in seg._child_containers:
+                for child_obj in getattr(seg, container_name):
+                    child_struct = self.create_struct_from_obj(child_obj)
+                    seg_struct[container_name].append(child_struct)
+
+        for group in bl.groups:
+            group_structure = self.create_struct_from_obj(group)
+            bl_struct["groups"].append(group_structure)
+
+            for container_name in group._child_containers:
+                for child_obj in getattr(group, container_name):
+                    if isinstance(child_obj, (ChannelView, RegionOfInterest)):
+                        child_struct = self.create_struct_from_view(child_obj)
+                        group_structure[container_name].append(child_struct)
+                    else:
+                        group_structure[container_name].append(id(child_obj))
 
-            for sptr in seg.spiketrains:
-                sptr_struct = self.create_struct_from_obj(sptr)
-                seg_struct['spiketrains'].append(sptr_struct)
+        scipy.io.savemat(self.filename, {"block": bl_struct}, oned_as="row")
 
-        scipy.io.savemat(self.filename, {'block': bl_struct}, oned_as='row')
+    def _get_matlab_value(self, ob, attrname):
+        units = None
+        if hasattr(ob, "_quantity_attr") and ob._quantity_attr == attrname:
+            units = ob.dimensionality.string
+            value = ob.magnitude
+        else:
+            try:
+                value = getattr(ob, attrname)
+            except AttributeError:
+                value = ob[attrname]
+            if isinstance(value, pq.Quantity):
+                units = value.dimensionality.string
+                value = value.magnitude
+            elif isinstance(value, datetime):
+                value = str(value)
+            elif isinstance(value, Mapping):
```

Comment:
Do we need an else here or do Quantity, datetime, and Mapping cover everything this could be? Maybe if it triggers the else we should put an error so we know in the future we are missing something.

Reply:
it would be 
```
else:
    value = value
```
i.e. all other values should just fall through, since MATLAB can handle them as is

Reply:
Sounds good. Makes sense.